### PR TITLE
Update AbstractControllerTestCase.php

### PR DESCRIPTION
**Pull request by SilverioMiranda**
This is the solution to $this->expectExceptionMessage not working in controller test like this :

```
$this->expectExceptionMessage("some exception message");
$this->dispatch("/", 'GET');
```

### CODE REVIEW

File: src/PHPUnit/Controller/AbstractControllerTestCase.php [hunk 1]
```diff
@@ -304,6 +304,14 @@ public function dispatch($url, $method = null, $params = [], $isXmlHttpRequest =
 
         $this->url($url, $method, $params);
         $this->getApplication()->run();
+        
+        if (true !== $this->traceError) {
+            return;
+        }
+        $exception = $this->getApplication()->getMvcEvent()->getParam('exception');
+        if ($exception instanceof \Exception) {
```
Comment (michalbundyra) [inline src/PHPUnit/Controller/AbstractControllerTestCase.php:312]: I would suggest to change here to have:
```php
if ($exception instanceof \Throwable || $exception instanceof \Exception)
```

so we can support all Throwables on PHP 7.0+.

**Review by michalbundyra — COMMENTED** [review]
I like that change. We need to add test for that functionality.
As suggested in the comment I would also add support for PHP 7.0 throwables.

**Comment by michalbundyra** [comment]
@SilverioMiranda Hm... after another look on it - this is BC Break. Please see failing tests. Currently we should use: `assertApplicationException` and we can also assert status code. Looks like that current functionality gives you more - as in the application you can have some error handlers etc, and you can assert the response etc.

**Comment by SilverioMiranda** [comment]
Yeah, after some research i agree that method assertApplicationException is the best alternative, i based my PR on source code from the company i work and they did that to handle exceptions, is a legacy code from 2010 and have alot of ad-hoc implementations

**Pull request closed without merging**